### DISCUSSION TITLE
What's the reason for the nested capabilities exposed on devices?

### DISCUSSION THREAD

**Discussion by tom-sherman** [Koenkk/zigbee2mqtt#5954] [General]
I'm referring to the the `exposes` field of the messages published to the `bridge/devices` topic.

Why do some capabilities exist at the top level of the exposes array, and some are nested in the first capability under a `features` key?

Here's how I'm currently modelling it in TypeScript:

https://github.com/tom-sherman/smart-home/blob/3128edb601d99e04c5adcc81ce114dae243c3437/services/zigbee2mqtt-controller-service/src/bridge-types.ts#L31-L47

**Comment by Koenkk** [comment]
Because some functionalities are really coupled to each other (like a light). Docs can be found here: https://www.zigbee2mqtt.io/information/exposes

**Reply by tom-sherman** [reply]
Thanks for the link, I had missed that somehow! So could a single device expose both a `switch` and a `light` for example?

EDIT: Silly question, this is obviously true by reading the page lol
